js-api-doc/compile.d.ts

```diff
@@ -3,28 +3,150 @@ import {RawSourceMap} from 'source-map-js';
 
 import {Options, StringOptions} from './options';
 
+/**
+ * The result of compiling Sass to CSS. Returned by [[compile]],
+ * [[compileAsync]], [[compileString]], and [[compileStringAsync]].
+ *
+ * @category Compile
+ */
 export interface CompileResult {
+  /**
+   * The generated CSS.
+   *
+   * Note that this *never* includes a `sourceMapUrl` comment—it's up to the
+   * caller to determine where to save the source map and how to link to it from
+   * the stylesheet.
+   */
   css: string;
+
+  /**
+   * The canonical URLs of all the stylesheets that were loaded during the
+   * Sass compilation. The order of these URLs is not guaranteed.
+   */
   loadedUrls: URL[];
+
+  /**
+   * The object representation of the source map that maps locations in the
+   * generated CSS back to locations in the Sass source code.
+   *
+   * This typically uses absolute `file:` URLs to refer to Sass files, although
+   * this can be controlled by having a custom [[Importer]] return
+   * [[ImporterResult.sourceMapUrl]].
+   *
+   * This is set if and only if [[Options.sourceMap]] is `true`.
+   */
   sourceMap?: RawSourceMap;
 }
 
-/** @hidden */
+/**
+ * Synchronously compiles the Sass file at `path` to CSS. If it succeeds it
+ * returns a [[CompileResult]], and if it fails it throws an [[Exception]].
+ *
+ * This only allows synchronous [[Importer]]s and [[CustomFunction]]s.
+ *
+ * @example
+ *
+ * ```js
+ * const sass = require('sass');
```

Comment:
nit: Should we use ESM here and throughout?

```suggestion
 * import * as sass from 'sass';
```

Reply:
Since afaik the latest version of Node still chokes on this without special configuration, I think it's not yet time to do it by default. People who are using ESM will know how to translate better than unfamiliar people. At some point we should make the switch, though.